imap_processing/codice/codice_l1a.py

```diff
@@ -79,143 +84,143 @@ def __init__(self, table_id: int, plan_id: int, plan_step: int, view_id: int):
-            np.arange(self.num_positions),
-            name="inst_az",
-            dims=["inst_az"],
-            attrs=cdf_attrs.get_variable_attributes("inst_az"),
-        )
-        spin_sector = xr.DataArray(
-            np.arange(self.num_spin_sectors),
-            name="spin_sector",
-            dims=["spin_sector"],
-            attrs=cdf_attrs.get_variable_attributes("spin_sector"),
-        )
-        esa_step = xr.DataArray(
-            np.arange(self.num_energy_steps),
-            name="esa_step",
-            dims=["esa_step"],
-            attrs=cdf_attrs.get_variable_attributes("esa_step"),
-        )
-
-        # Create the dataset to hold the data variables
+        # Create the main dataset to hold all the variables
         dataset = xr.Dataset(
-            coords={
-                "epoch": epoch,
-                "inst_az": inst_az,
-                "spin_sector": spin_sector,
-                "esa_step": esa_step,
-            },
-            attrs=cdf_attrs.get_global_attributes(self.dataset_name),
+            coords=self.coords,
+            attrs=self.cdf_attrs.get_global_attributes(self.dataset_name),
         )
 
         # Create a data variable for each counter
         for variable_data, variable_name in zip(self.data, self.variable_names):
-            # Data arrays are structured depending on the instrument
-            if self.instrument == "lo":
-                variable_data_arr = np.array(variable_data).reshape(
-                    (
-                        1,
-                        self.num_positions,
-                        self.num_spin_sectors,
-                        self.num_energy_steps,
-                    )
-                )
-                dims = ["epoch", "inst_az", "spin_sector", "esa_step"]
-            elif self.instrument == "hi":
-                variable_data_arr = np.array(variable_data).reshape(
-                    (
-                        1,
-                        self.num_energy_steps,
-                        self.num_positions,
-                        self.num_spin_sectors,
-                    )
-                )
-                dims = ["epoch", "esa_step", "inst_az", "spin_sector"]
+            # Reshape to 4 dimensions to allow for epoch dimension
+            reshaped_variable_data = np.expand_dims(variable_data, axis=0)
 
             # Get the CDF attributes
             cdf_attrs_key = (
                 f"{self.dataset_name.split('imap_codice_l1a_')[-1]}-{variable_name}"
             )
-            attrs = cdf_attrs.get_variable_attributes(cdf_attrs_key)
+            attrs = self.cdf_attrs.get_variable_attributes(cdf_attrs_key)
 
             # Create the CDF data variable
             dataset[variable_name] = xr.DataArray(
-                variable_data_arr,
+                reshaped_variable_data,
                 name=variable_name,
-                dims=dims,
+                dims=self.dims,
                 attrs=attrs,
             )
```

Comment:
I was following pretty good up until this. I am not following how `self.data` is associated with `self.variable_names`.

Reply:
`self.data` is a list of multi-dimensional numpy arrays, with each element of the list being the data for a given CDF variable. THe `self.variable_names` provides the list of CDF variable names (e.g. `h`, `he`, `cno`, etc.) to associate with each element of `self.data`.

This has got me realizing that I should change some of the variable names I am using in `constants.py` from something like `LO_SW_ANGULAR_NAMES` to `LO_SW_ANGULAR_VARIABLE_NAMES` to be more explicit.